packages/commonwealth/client/scripts/views/pages/CommunityGroupsAndMembers/Groups/common/GroupForm/Allowlist/Allowlist.tsx

```diff
@@ -0,0 +1,272 @@
+];
+
+const MEMBERS_PER_PAGE = 10;
+
+const Allowlist = ({
+  allowedAddresses,
+  setAllowedAddresses,
+}: AllowListProps) => {
+  useUserActiveAccount();
+
+  const [searchFilters, setSearchFilters] = useState({
+    searchText: '',
+    groupFilter: 'all-community',
+  });
+
+  const [currentPage, setCurrentPage] = useState<number>(1);
+
+  const { data: memberships } = useRefreshMembershipQuery({
+    communityId: app.activeChainId(),
+    address: app?.user?.activeAccount?.address,
+    apiEnabled: !!app?.user?.activeAccount?.address,
+  });
+
+  const isStakedCommunity = !!app.config.chains.getById(app.activeChainId())
+    .namespace;
+
+  const tableState = useCWTableState({
+    columns: tableColumns(isStakedCommunity),
+    initialSortDirection: APIOrderDirection.Desc,
+  });
+
+  const debouncedSearchTerm = useDebounce<string>(
+    searchFilters.searchText,
+    500,
+  );
+
+  const memoizedAddresses = useMemo(() => {
+    return allowedAddresses;
+    // we disable the exhaustive-deps because we don't want to refresh on changed allowedAddresses because it will
+    // update the displayed list while the boxes are being checked which is a bit jarring from a UI perspective
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [searchFilters]);
+
+  const { groups, isLoadingMembers, members } = useMemberData({
+    tableState,
+    groupFilter: searchFilters.groupFilter,
+    debouncedSearchTerm,
+    memberships,
+    membersPerPage: MEMBERS_PER_PAGE,
+    page: currentPage,
+    allowedAddresses: memoizedAddresses,
+  });
+
+  const handlePageChange = (_e, page: number) => {
+    setCurrentPage(page);
+  };
+  const formattedMembers: Member[] = useMemo(() => {
+    return (
+      (members?.results?.map((p) => ({
+        id: p.id,
+        avatarUrl: p.avatar_url,
+        name: p.profile_name || 'Anonymous',
+        role: p.roles[0],
+        groups: (p.group_ids || [])
+          .map(
+            (groupId) =>
+              (groups || []).find((group) => group.id === groupId)?.name,
+          )
+          .filter(Boolean)
+          .sort((a, b) => a.localeCompare(b)),
+        stakeBalance: p.addresses[0].stake_balance,
+        address: p.addresses[0].address,
+      })) as Member[]) || []
+    );
+  }, [groups, members?.results]);
+
+  const handleCheckboxChange = (address: string) => {
+    if (allowedAddresses.includes(address)) {
+      setAllowedAddresses((prevItems) =>
+        prevItems.filter((item) => item !== address),
+      );
+    } else {
+      setAllowedAddresses((prevItems) => [...prevItems, address]);
+    }
+  };
+
+  const extraColumns = (member: Member) => {
+    return {
+      address: {
+        sortValue: member.address,
+        customElement: (
+          <div className="table-cell">
+            {formatAddressShort(member.address, 5, 6)}
+          </div>
+        ),
+      },
+    };
+  };
+
+  console.log(currentPage);
+
+  return (
+    <section className="AllowList">
```

Comment:
file is Allowlist, className is AllowList, props are AllowListProps etc

search all occurrences of AllowList and change it to Allowlist